### PR TITLE
fix(DB/quest_greeting): Populate quest_greeting table

### PR DESCRIPTION
**Pull request by ZhengPeiRu21**
## Changes Proposed:
-  Add quest_greeting data from TrinityCore and Vmangos

## SOURCE:

TrinityCore and Vmangos

## Tests Performed:

- Several NPCs spot-test, all seem to work as intended:
Capture

## How to Test the Changes:

1. Go to any NPCs mentioned in PR when multiple quest or gossip available
2. Test that Quest Greeting is appear correctly

## Known Issues and TODO List:

There are some NPCs that have both quest_greeting text and a gossip menu with gossip flag. This is consistent with Vmangos and TrinityCore, so I think it is okay, but it does seem like there would be no reason to have quest_greeting text in these case because it would be override by the gossip_menu text. An example is creature 240.

I have also not added the quest_greeting_locale data yet. Both TrinityCore and Vmangos are very incomplete with locale data, so I think we will need a separate sniff effort to collect this.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1646669102474392500.sql [hunk 1]
```diff
@@ -0,0 +1,200 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1646669102474392500');
+
+DELETE FROM `quest_greeting` WHERE `ID` BETWEEN 234 AND 186420;
```
Comment (acidmanifesto) [inline data/sql/updates/pending_db_world/rev_1646669102474392500.sql:3]: this makes me a bit nervous as that is a pretty huge range between. are we certain you got everything between this range in the insert below?

**Review by acidmanifesto — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1646669102474392500.sql (hunk 1 above)
Comment (ZhengPeiRu21) [reply data/sql/updates/pending_db_world/rev_1646669102474392500.sql:3]: Yes, the table is currently empty. It was just introduced by https://github.com/azerothcore/azerothcore-wotlk/pull/10526 this past weekend. However, if we would prefer I can have it only remove the entries explicitly added. Should I make that change?

Comment (ZhengPeiRu21) [reply data/sql/updates/pending_db_world/rev_1646669102474392500.sql]: I have updated PR to only remove the IDs only specifically added beneath. Thank you for the suggestion! I think it is a good change.

**Review by acidmanifesto — APPROVED** [review]

**Review by Kitzunu — APPROVED** [review]

**Comment by Bogir** [comment]
Can we find localized strings in data-files extracted from localized clients?

**Comment by ZhengPeiRu21** [comment]
> Can we find localized strings in data-files extracted from localized clients?

From what I can know, I think quest greeting data is not in client data and needs to be sniffed, but I could be mistake.

**Pull request merged**